RecoHI/HiEvtPlaneAlgos/interface/HiEvtPlaneFlatten.h

```diff
@@ -202,30 +202,30 @@ class HiEvtPlaneFlatten {
   double getVtxMax() const {return minvtx_+nvtxbins_*delvtx_;}
   int getNcent() const {return hbins_;}
 
-  double getX(int bin) const {return flatX_[bin];}
-  double getY(int bin) const {return flatY_[bin];}
-  double getXoff(int bin) const {return xoff_[bin];}
-  double getYoff(int bin) const {return yoff_[bin];}
-  double getXoffDB(int bin) const {return xoffDB_[bin];}
-  double getYoffDB(int bin) const {return yoffDB_[bin];}
-  double getXYoffcnt(int bin) const {return xyoffcnt_[bin];}
-  double getXYoffmult(int bin) const {return xyoffmult_[bin];}
-  double getPt(int bin) const {return pt_[bin];}
-  double getPt2(int bin) const {return pt2_[bin];}
-  double getPtDB(int bin) const {return ptDB_[bin];}
-  double getPt2DB(int bin) const {return pt2DB_[bin];}
-  double getPtcnt(int bin) const {return ptcnt_[bin];}
-  double getXDB(int bin)  const {return flatXDB_[bin];}
-  double getYDB(int bin)  const {return flatYDB_[bin];}
-
-
-  double getCnt(int bin)  const {return flatCnt_[bin];}
-  void setXDB(int indx, double val) {flatXDB_[indx]=val;}
-  void setYDB(int indx, double val) {flatYDB_[indx]=val;}
-  void setXoffDB(int indx, double val) {xoffDB_[indx]=val;}
-  void setYoffDB(int indx, double val) {yoffDB_[indx]=val;}
-  void setPtDB(int indx, double val) {ptDB_[indx]=val;}
-  void setPt2DB(int indx, double val) {pt2DB_[indx]=val;}
+  double getX(unsigned int bin) const {return flatX_[bin];}
+  double getY(unsigned int bin) const {return flatY_[bin];}
+  double getXoff(unsigned int bin) const {return xoff_[bin];}
+  double getYoff(unsigned int bin) const {return yoff_[bin];}
+  double getXoffDB(unsigned int bin) const {return xoffDB_[bin];}
+  double getYoffDB(unsigned int bin) const {return yoffDB_[bin];}
+  double getXYoffcnt(unsigned int bin) const {return xyoffcnt_[bin];}
+  double getXYoffmult(unsigned int bin) const {return xyoffmult_[bin];}
+  double getPt(unsigned int bin) const {return pt_[bin];}
+  double getPt2(unsigned int bin) const {return pt2_[bin];}
+  double getPtDB(unsigned int bin) const {if(bin<MAXCUTOFF) {return ptDB_[bin];} else {return 0.;}}
+  double getPt2DB(unsigned int bin) const {if(bin<MAXCUTOFF) {return pt2DB_[bin];} else {return 0.;}}
+  double getPtcnt(unsigned int bin) const {return ptcnt_[bin];}
+  double getXDB(unsigned int bin)  const {return flatXDB_[bin];}
+  double getYDB(unsigned int bin)  const {return flatYDB_[bin];}
+
+
+  double getCnt(unsigned int bin)  const {return flatCnt_[bin];}
+  void setXDB(unsigned int indx, double val) {flatXDB_[indx]=val;}
+  void setYDB(unsigned int indx, double val) {flatYDB_[indx]=val;}
+  void setXoffDB(unsigned int indx, double val) {xoffDB_[indx]=val;}
+  void setYoffDB(unsigned int indx, double val) {yoffDB_[indx]=val;}
+  void setPtDB(unsigned int indx, double val) {ptDB_[indx]=val;}
+  void setPt2DB(unsigned int indx, double val) {pt2DB_[indx]=val;}
```

Comment:
OK, as long as there is protection against out-of-bounds errors, further changes aren't essential.